### PR TITLE
feat: aggregate AccountIds staking data

### LINKED ISSUE #2615: DeFi - overview and opportunities list pages
### Overview

Currently the defi overview page assumes a single account when displaying summaries, and we only support opportunities for EVM and CosmosSDK based chains.

### References and additional details

* We should show aggregated balances for opportunities across assets on the overview page, and opportunities list page
* Individual opportunities are handled in separate tickets

### Acceptance Criteria

* Ensure all selectors are aggregating across multiple accounts for the overview and opportunities list pages

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

2 days

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR bring support for aggregation of staking data from multiple AccountIds.

To achieve this, selectors that were previously returning / consuming `StakingDataByValidatorId | null` now return / expect a `(StakingDataByValidatorId | null)[]` collection - no matter whether or not we're selecting data for a specific AccountId (1-length array) or for all AccountIds for a given AssetId.

Consumers are then able to flatten this collection to get an aggregated staking data (e.g validator delegations) for a specific AccountId.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/blob/develop/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #2615 

## Risk

- Places where we previously had aggregated data could be borked and show the data for a single account
- Places where we previously had data for a single account could show the sum of all accounts instead

## Testing

- Add one or more Cosmos accounts - both account 0 and 1(+) should have staking data
- In the Cosmos **asset** page Staking section, each row should show the sum of your delegations for that validator across all accounts
- In the Cosmos **account** page Staking section, each row should show the value of your delegations for that validator for that specific account
- In the DeFi overview section, each Cosmos card should show the sum of your delegations for that validator across all accounts
- In the DeFi earn section, each Cosmos row should show the sum of your delegations for that validator across all accounts

### Engineering

- Implementation prevents musculoskeletal disorders in the workplace

### Operations

- See risks and top-level notes

#### Account 0 page

image

#### Account 1 page

image

#### Asset Page

image

#### DeFi Overview cards

image

#### DeFi Earn rows

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3051** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3051" target="_blank">Graphite</a>  👈
    * **PR #3052** <a href="https://app.graphite.dev/github/pr/shapeshift/web/3052" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/3051?utm_source=stack-comment).

File: src/state/slices/portfolioSlice/selectors.ts [hunk 1]
```diff
@@ -419,21 +424,53 @@ export const selectPortfolioCryptoHumanBalanceByFilter = createSelector(
...
+export const selectPortfolioAccountIdsByAssetId = createCachedSelector(
+  selectPortfolioAccountIds,
+  selectAssetIdParamFromFilter,
+  (accountIds, assetId): AccountId[] => {
+    // early return for scenarios where assetId is not available yet
+    if (!assetId) return []
+    const { chainId } = fromAssetId(assetId)
+    return accountIds.filter(accountId => fromAccountId(accountId).chainId === chainId)
+  },
+)((_accountIds, { assetId }) => assetId ?? 'undefined')
```
Comment (gomesalexandre) [inline src/state/slices/portfolioSlice/selectors.ts:446]: This is just shuffled around

File: src/state/slices/portfolioSlice/selectors.ts [hunk 2]
```diff
@@ -630,27 +670,6 @@ export const selectPortfolioAssets = createSelector(
...
-  selectPortfolioAccountIds,
-  selectAssetIdParamFromFilter,
-  (accountIds, assetId): AccountId[] => {
-    // early return for scenarios where assetId is not available yet
-    if (!assetId) return []
-    const { chainId } = fromAssetId(assetId)
-    return accountIds.filter(accountId => fromAccountId(accountId).chainId === chainId)
-  },
-)((_accountIds, { assetId }) => assetId ?? 'undefined')
-
```
Comment (gomesalexandre) [inline src/state/slices/portfolioSlice/selectors.ts:653]: This is only shuffled around - not deleted code

File: src/state/slices/portfolioSlice/selectors.ts [hunk 3]
```diff
@@ -1324,15 +1359,23 @@ export const selectStakingOpportunitiesDataFull = createDeepEqualOutputSelector(
         // Delegated/Redelegated + Undelegation
         totalDelegations,
         // Rewards at 0 index: since we normalize staking data, we are guaranteed to have only one entry for the validatorId + assetId combination
-        rewards: stakingDataByValidator?.[validatorId]?.[assetId]?.rewards?.[0]?.amount ?? '0',
+        rewards: stakingDataByValidator
+          .reduce((acc, currentStakingDataByValidator) => {
+            acc = acc.plus(
+              currentStakingDataByValidator?.[validatorId]?.[assetId]?.rewards?.[0]?.amount ?? '0',
```
Comment (gomesalexandre) [inline src/state/slices/portfolioSlice/selectors.ts:1365]: Note: This isn't a user-facing change, we currently don't have anywhere in the app where aggregated rewards should be displayed, and effectively only reach code paths where we wrap a single AccountId

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**